File: php/testcode/ssti_035.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ssti_twig_sandbox_enabled
function ssti035(BenchmarkRequest $req): BenchmarkResponse {
    $twig = new Twig\Environment(new Twig\Loader\FilesystemLoader('/views'));
    $sandbox = new Twig\Extension\SandboxExtension(new Twig\Sandbox\SecurityPolicy());
    $twig->addExtension($sandbox);
    $twig->enableSandbox(); // vuln-code-snippet safe-line php_ssti_twig_sandbox_enabled
    $html = $twig->render($req->param('tpl'), []);
    return BenchmarkResponse::ok($html);
}
// vuln-code-snippet end php_ssti_twig_sandbox_enabled

File: ruby/testcode/ssti_035.rb
require_relative 'shared'

begin
  require 'haml'
rescue LoadError
end

# vuln-code-snippet start ruby_ssti_haml_fixed_file
def handler(req)
  output = Haml::Engine.new(File.read('templates/email.haml')).render(Object.new, name: req.param('name')) # vuln-code-snippet safe-line ruby_ssti_haml_fixed_file
  BenchmarkResponse.html(output)
end
# vuln-code-snippet end ruby_ssti_haml_fixed_file
